File: TH_Buoi2/ViDu_VongLapDK_O_logn_Bai2.cpp
#include <iostream>
#include <vector>

int binary_search(const std::vector<int>& arr, int target) {
    int left = 0, right = arr.size() - 1;
    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1;
}

int main() {
    std::vector<int> arr;
    for (int i = 1; i <= 16; ++i) {
        arr.push_back(i);
    }
    std::cout << binary_search(arr, 10) << std::endl; // Output: 9
    return 0;
}

File: TH_Buoi2/ViDu_VongLapDK_O_logn_Bai2.jl
function binary_search(arr, target)
    left, right = 1, length(arr)
    while left <= right
        mid = div(left + right, 2)
        if arr[mid] == target
            return mid
        elseif arr[mid] < target
            left = mid + 1
        else
            right = mid - 1
        end
    end
    return -1
end

arr = collect(1:16) # [1, 2, 3, ..., 16]
println(binary_search(arr, 10)) # Output: 10